Formula/c/chakra.rb

```diff
@@ -36,16 +36,14 @@ class Chakra < Formula
   depends_on "cmake" => :build
   depends_on "icu4c"
 
+  uses_from_macos "llvm" => :build
   uses_from_macos "python" => :build
 
-  on_linux do
-    # Currently requires Clang, but fails with LLVM 16.
-    depends_on "llvm@15" => :build
+  fails_with :gcc do
+    cause "requires Clang, see https://github.com/chakra-core/ChakraCore/issues/2038"
   end
 
   def install
-    ENV.clang if OS.linux? # Currently fails to build with LLVM 16.
-
     # Use ld_classic to work around 'ld: Assertion failed: (0 && "lto symbol should not be in layout")'
     ENV.append "LDFLAGS", "-Wl,-ld_classic" if DevelopmentTools.clang_build_version >= 1500
```

Comment:
```suggestion
    ENV.append "LDFLAGS", "-Wl,-ld_classic" if OS.mac? && DevelopmentTools.clang_build_version >= 1500
```
Probably don't want to pass this on Linux.

Reply:
Could add, but it should be a skipped either way as `clang_build_version` is only set on macOS due to regex. `clang_version` would be used on both.